include/deal.II/numerics/data_out_dof_data.templates.h

```diff
@@ -83,6 +83,98 @@ namespace internal
 
 
 
+    /**
+     * Generate evalution points on a simplex with arbitrary number of
+     * subdivisions.
+     */
+    template <int dim>
+    inline std::vector<Point<dim>>
+    generate_simplex_evaluation_points(const unsigned int n_subdivisions)
+    {
+      (void)n_subdivisions;
+
+      Assert(false, ExcNotImplemented());
+
+      return {};
+    }
+
+
+
+    /**
+     * Helper function to create evaluation points recursivly.
+     */
+    inline void
+    generate_simplex_evaluation_points_recursivly(
+      const std::vector<Point<2>> &bounding_vertices,
+      const unsigned int           n_subdivisions,
+      std::vector<Point<2>> &      evalution_points)
+    {
+      if (n_subdivisions == 0)
+        {
+          evalution_points.push_back(bounding_vertices[0]);
+          return;
+        }
+
+      for (const auto &p : bounding_vertices)
+        evalution_points.push_back(p);
+
+      if (n_subdivisions == 1)
+        return;
+
+      const auto process = [&](const Point<2> &p0, const Point<2> &p1) {
+        std::vector<Point<2>> line_points;
+
+        for (unsigned int i = 1; i < n_subdivisions; ++i)
+          line_points.push_back(p0 + (p1 - p0) / n_subdivisions * i);
+
+        evalution_points.insert(evalution_points.end(),
+                                line_points.begin(),
+                                line_points.end());
+
+        return line_points;
+      };
+
+      const auto line_points_0 =
+        process(Point<2>(bounding_vertices[0]), Point<2>(bounding_vertices[1]));
```

Comment:
```suggestion
        process(bounding_vertices[0], bounding_vertices[1]);
```
Same for the lines below